client/src/actions/actionsUser/usersActions.js

```diff
@@ -101,23 +101,45 @@ export const newAccount = (user, pass, email) =>{
 }
 export const login = (user, pass) =>{
   return function(dispatch) {
+  //  console.log(user);
     dispatch(fetchUserRequest())
-    console.log(user);
-    console.log(pass);
      fetch('http://localhost:55555/user/login', {
       method: 'POST',
       body:JSON.stringify({'username': user, 'password':pass})
       })
-     .then( res => {
-      console.log(res);
-      dispatch(fetchUserSucess(res.data))
+     .then(function(response) {
+       if(false) {
```

Comment:
This is because from endpoint you are not returning something useful here

[User#validateCredentials](https://github.com/saiema/ProgramRepairGolf-DOSE2019/blob/master/src/main/java/unrc/dose/User.java#L265) returns a boolean and from the endpoint [/login](https://github.com/saiema/ProgramRepairGolf-DOSE2019/blob/master/src/main/java/unrc/dose/UserEndpoint.java#L105) you are only returning this boolean, so what you have to do is to respond with the correct status code when correspond:

1. code 200 when the login is value
3. code 401 when unauthorized

```
     String username = bodyParams.get("username")
     String pass = bodyParams.get("password");
     if(User.validateCredentials(username, password)) {
       response.status(200);
     } else {
       response.status(401);
     }
```

in this way when everything goes ok you will use `then` function here but when you are unauthorized you will run the `catch` function

Reply:
ok I understand. I will modify the file now and pull up the changes.